src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -779,77 +780,84 @@ void DlgPrefMixer::slotUpdate() {
 // Draw the crossfader curve graph. Only needs to get drawn when a change
 // has been made.
 void DlgPrefMixer::drawXfaderDisplay() {
-    constexpr int kGrindXLines = 4;
-    constexpr int kGrindYLines = 6;
-
-    int frameWidth = graphicsViewXfader->frameWidth();
-    int sizeX = graphicsViewXfader->width() - 2 * frameWidth;
-    int sizeY = graphicsViewXfader->height() - 2 * frameWidth;
-
-    // Initialize Scene
+    // Initialize or clear scene
     if (m_pxfScene) {
         m_pxfScene->clear();
     } else {
         m_pxfScene = make_parented<QGraphicsScene>(this);
         // The size of the QGraphicsView doesn't change so we need to do this only once
+        graphicsViewXfader->setLineWidth(1); // frame width
+        int sizeX = graphicsViewXfader->width() - 2;
+        int sizeY = graphicsViewXfader->height() - 2;
         m_pxfScene->setSceneRect(0, 0, sizeX, sizeY);
         m_pxfScene->setBackgroundBrush(Qt::black);
+        graphicsViewXfader->setRenderHints(QPainter::Antialiasing);
         graphicsViewXfader->setScene(m_pxfScene);
     }
 
     // Initialize QPens
-    QPen gridPen(Qt::green);
-    QPen graphLinePen(Qt::white);
+    QPen gridPen(Qt::gray);
+    QPen gainPen(Qt::white);
+    QPen totalPen(Qt::red);
+    // In conjunction with anti-aliasing this gives smooth, solid lines.
+    totalPen.setWidth(2);
+    gainPen.setWidth(2);
+    // For some reason grid lines also appear 2px wide, with the nice side effect
+    // that gain curves now intersect 'exactly'at the grid center line.
+
+    const int sceneW = static_cast<int>(m_pxfScene->width());
+    const int sceneH = static_cast<int>(m_pxfScene->height());
 
     // Draw grid
-    for (int i = 1; i < kGrindXLines; i++) {
-        m_pxfScene->addLine(
-                QLineF(0, i * (sizeY / kGrindXLines), sizeX, i * (sizeY / kGrindXLines)), gridPen);
+    const int kGridHLines = 3;
+    const int kGridVLines = 5;
```

Comment:
should this be constexpr static and at the top of the file to follow the usual pattern?